### PR TITLE
Cleanup chart labels

### PR DESCRIPTION
**Pull request by bpblanken**
Some miscellaneous cleanup to improve the consistency of the labeling as we add another chart.

### CODE REVIEW

File: charts/hail-search/templates/_helpers.tpl [hunk 1]
```diff
@@ -1,25 +1,13 @@
-{{/*
-Expand the name of the chart.
-*/}}
-{{- define "hail-search.name" -}}
```
Comment (bpblanken) [inline charts/hail-search/templates/_helpers.tpl:4]: I got rid of `nameOverride` and `fullNameOverride` and `name` just to simplify the logic.  I think it makes sense to allow external users to override the 'release' but the applications should still be named what they're named.

File: charts/hail-search/templates/_helpers.tpl [hunk 2]
```diff
@@ -1,25 +1,13 @@
...
-{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
-{{- else }}
-{{- $name := default .Chart.Name .Values.nameOverride }}
-{{- if contains $name .Release.Name }}
+{{- if contains .Chart.Name .Release.Name }}
 {{- .Release.Name | trunc 63 | trimSuffix "-" }}
 {{- else }}
-{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
-{{- end }}
+{{- printf "%s-%s" .Release.Name .Chart.Name | trunc 63 | trimSuffix "-" }}
```
Comment (bpblanken) [inline charts/hail-search/templates/_helpers.tpl:10]: release name is passed in [here](https://github.com/broadinstitute/seqr-private/blob/master/deployments/seqr/dev/seqr.yaml#L16)

File: charts/seqr/values.yaml [hunk 3]
```diff
@@ -28,6 +24,9 @@ serviceAccount:
 # -- A dictionary of annotations to add to the seqr Deployment
 podAnnotations: {}
 
+nodeSelector:
```
Comment (bpblanken) [inline charts/seqr/values.yaml:27]: We're using nodeSelector already.

File: charts/seqr/values.yaml [hunk 4]
```diff
@@ -84,9 +83,6 @@ affinity: |-
               {{- include "seqr.selectorLabels" . | nindent 12 }}
           topologyKey: "kubernetes.io/hostname"
 
-# -- Pod tolerations for the seqr Deployment
-tolerations: []
```
Comment (bpblanken) [inline charts/seqr/values.yaml:88]: We're not using tolerations... I think it's easier to support only `nodeSelector` and `affinity`?

File: charts/seqr/templates/migration.yaml [hunk 5]
```diff
@@ -1,39 +0,0 @@
-{{- if .Values.run_seqr_database_migration }}
```
Comment (bpblanken) [inline charts/seqr/templates/migration.yaml:1]: this is out of date and unused, I think better to just re-write with the current `Job` setup that we're using for check-new-samples when we want to enable?

File: charts/seqr/templates/deployment.yaml [hunk 6]
```diff
@@ -2,15 +2,9 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: seqr
+  name: {{ include "seqr.fullname" . }}
   labels:
-    name: seqr
-    deployment: {{ .Release.Name }}
     {{- include "seqr.labels" . | nindent 4 }}
-  {{- if .Values.deploymentAnnotations }}
```
Comment (bpblanken) [inline charts/seqr/templates/deployment.yaml:10]: doesn't exist anywhere.

File: charts/hail-search/templates/_helpers.tpl (hunk 1 above)
Comment (sjahl) [reply charts/hail-search/templates/_helpers.tpl:4]: Yeah, I'm fine with this -- these tpls that get generated with the helm create command are pretty broad, intending to cover pretty much any use case, so narrowing this a bit seems fine.

File: charts/hail-search/templates/_helpers.tpl [hunk 7]
```diff
@@ -46,6 +34,6 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
 Selector labels
 */}}
 {{- define "hail-search.selectorLabels" -}}
-app.kubernetes.io/name: {{ include "hail-search.name" . }}
+app.kubernetes.io/name: {{ include "hail-search.fullname" . }}
```
Comment (sjahl) [inline charts/hail-search/templates/_helpers.tpl:37]: I would proceed with caution when updating selectorLabels -- I believe that it is an immutable field on the k8s API, and changing what goes in here will require deleting the running deployment, and redeploying.

I don't know how many folks out there are actually using these helm charts besides us, so we may no tneed to worry too much, but figured I'd mention it.

File: charts/seqr/templates/deployment.yaml (hunk 6 above)
Comment (sjahl) [reply charts/seqr/templates/deployment.yaml:10]: It's a fairly typical pattern in helm charts to provide empty annotations and tolerations, as these are things that end-deployers frequently need to change. Maybe it's better to provide an empty `{}` like the podAnnotations in the default values.yaml?

File: charts/seqr/values.yaml (hunk 4 above)
Comment (sjahl) [reply charts/seqr/values.yaml:88]: Same thought as the deploymentAnnotations here -- it's kind of hard to predict what tolerations an end user might need, so the general practice is provide an empty default that allows people to set arbitrary tolerations.

e.g. Azure kubernetes service might taint whatever they call spot nodes, and require deployments to tolerate that if you want to run the workload there... but we don't necessarily want to encode that in this chart, or default to tolerating that.

**Review by sjahl — COMMENTED** [review]
Just a few comments inline on some of the annotations / tolerations changes. tl;dr, the spirit with the empty defaults on those is that these are things that users need to manipulate pretty often.

File: charts/hail-search/templates/_helpers.tpl (hunk 7 above)
Comment (bpblanken) [reply charts/hail-search/templates/_helpers.tpl:37]: 👍 good call out.  I think it'll just impact us at the moment, but this is definitely a change for us from `hail-search` to `seqr-hail-search` (including the "release name").

File: charts/seqr/templates/deployment.yaml (hunk 6 above)
Comment (bpblanken) [reply charts/seqr/templates/deployment.yaml:10]: 👍 that's doable! 

I have this overwhelming question (kind of across this repo) about how we should go about supporting features that we ourselves don't use.  This is a good example where we haven't implemented `deploymentAnnotations` elsewhere (I guess by accident?) or tested it and it isn't provided by default by the `helm create`.   

I guess the best plan of action is to actually go through and make sure both `deploymentAnnotations` and `tolerations` are appropriately implemented on all charts/pods... which they definitely are not right now.

File: charts/hail-search/templates/_helpers.tpl (hunk 2 above)
Comment (bpblanken) [reply charts/hail-search/templates/_helpers.tpl:10]: @sjahl actually.... for the sake of dogfooding this, how would we feel about changing the .Release.Name in our deployment to `broad`?  That'd be consistent with what we've put in our [readme](https://github.com/broadinstitute/seqr-helm/pull/54/files#diff-2d0cae1998d1122082f41d3a4aaace9eab74aa392851ddb1cb86450c1a79c408R58) and what we're likely to instruct external users to do when the process gets solidified.  

I know that doing that will for sure will require [this env var](https://github.com/broadinstitute/tgg-helm/blob/main/charts/lib-broad-seqr/values.yaml#L74) to change, and maybe more that are yet unknown?

**Comment by bpblanken** [comment]
I decided to be even more aggressive after a conversation and have completely removed `seqr.fullname` in favor of `{{ .Chart.Name}}.  Labels now look like:

```
  labels:
    helm.sh/chart: seqr-1.0.8
    app.kubernetes.io/name: seqr
    app.kubernetes.io/instance: broad
    app.kubernetes.io/version: "ed7a1736952b11246f3d40578f01284c90528528"
    app.kubernetes.io/managed-by: Helm
```

when I run `helm template broad charts/seqr`.

File: charts/seqr/templates/deployment.yaml (hunk 6 above)
Comment (sjahl) [reply charts/seqr/templates/deployment.yaml:10]: Yeah, in general I feel like it's fine to draw the line somewhere... At a certain point, if people want to do stuff that's way outside what we anticipate, they should probably be building their own deployment.

But, I think a good minimum is to ensure that people can pass in arbitrary labels, annotations, and tolerations on most of the chart's resources. I think a lot of people will end up needing to provide those at some point for one reason or another, and having the settings will keep them from needing to fork the chart. So, I guess a good barometer here is asking "_should_ someone have to fork the chart to do this?"

- e.g. labels, annotations, adding name suffixes to things, probably not?
- e.g. "I need to make seqr use MongoDB for [webscale](https://www.youtube.com/watch?v=b2F-DItXtZs)", probably time for your own chart

**Review by sjahl — APPROVED** [review]
Thanks -- switching to `.Chart.Name` seems reasonable!

File: charts/seqr/templates/deployment.yaml (hunk 6 above)
Comment (bpblanken) [reply charts/seqr/templates/deployment.yaml:10]: +1 this makes sense.  

My gut on this is that we want to make this as simple and dumb as possible to try to emulate a one-click install for people who don't know anything about helm or kubernetes.  I'd love if we were able to support the "I need this pod on this machine type" with just `nodeSelector` and `affinity`, leaving out `tolerations`, but I can see the use for supporting it.  Hopefully with the better unit testing we can at least get some coverage over those features so they're not just in the way.

**Pull request merged**